openml/runs/functions.py

```diff
@@ -571,7 +594,8 @@ def _create_run_from_xml(xml):
 
     files = dict()
     evaluations = dict()
-    detailed_evaluations = defaultdict(lambda: defaultdict(dict))
+    fold_evaluations = defaultdict(lambda: defaultdict(dict))
+    sample_evaluations = defaultdict(lambda: defaultdict(lambda: defaultdict(dict)))
```

Comment:
It's all about assumptions of the datastructure. We can always know for sure that a fold_evaluation in a 3-deep dict, and a sample_evaluation is a 4-deep dict (if these have entrees).